### PR TITLE
remove dependency on docker-compose when starting up archgw

### PR DESCRIPTION
**Pull request by adilhafeez**
fixes https://github.com/katanemo/archgw/issues/226

### CODE REVIEW

File: arch/tools/cli/core.py [hunk 1]
```diff
@@ -1,40 +1,73 @@
...
+        ports={
+            "10000/tcp": 10000,
+            "10001/tcp": 10001,
+            "11000/tcp": 11000,
+            "12000/tcp": 12000,
+            "19901/tcp": 19901,
+        },
+        volumes={
+            f"{arch_config_file}": {
+                "bind": "/app/arch_config.yaml",
```
Comment (salmanap) [inline arch/tools/cli/core.py:38]: does this require someone to checkout main to run archgw?

File: arch/tools/cli/core.py [hunk 2]
```diff
@@ -1,40 +1,73 @@
...
 from huggingface_hub import snapshot_download
+from dotenv import dotenv_values
+
 
 log = getLogger(__name__)
 
+dot_env = dotenv_values(".env")
+
+
+def start_archgw_docker(client, arch_config_file):
```
Comment (salmanap) [inline arch/tools/cli/core.py:24]: where does the CLI need to be to start arch?

File: arch/tools/build_cli.sh [hunk 3]
```diff
@@ -1,23 +1,4 @@
 #!/bin/bash
 
-# Define paths
-source_schema="../arch_config_schema.yaml"
```
Comment (salmanap) [inline arch/tools/build_cli.sh:4]: Its' unclear to me how the config_generator.py gets access to this schema to validate the arch_config from the developer. We have to have the cli run independently - meaning it should not require a fork of main. For those who want to `archgw build` from source, its necessary for them to fork main and run locally.

**Review by salmanap — CHANGES_REQUESTED** [review]
I like the removal of the docker compose dependency - but I have some other questions around the usage of arch_config_schema.yaml

File: arch/tools/build_cli.sh (hunk 3 above)
Comment (adilhafeez) [reply arch/tools/build_cli.sh:4]: schema file is burned into docker image - see here https://github.com/katanemo/archgw/blob/main/arch/Dockerfile#L26

File: arch/tools/cli/core.py (hunk 2 above)
Comment (adilhafeez) [reply arch/tools/cli/core.py:24]: here https://github.com/katanemo/archgw/pull/305/files#diff-9fb218171a8bf47ce0a494ac2ccdf5052d2525f6a0223b0ab327e942b3c48c87R132

File: arch/tools/cli/core.py (hunk 1 above)
Comment (adilhafeez) [reply arch/tools/cli/core.py:38]: no, its part of archgw cli

**Review by salmanap — APPROVED** [review]

**Pull request merged**